src/items/generics.md

```diff
@@ -83,6 +83,58 @@ where
 }
 ```
 
+## Free variables and concrete items and types
+
+The *free variables* of an item or type are the type and lifetime variables
+not bound therein. These variables may be substituted for other types or lifetimes. A *concrete item* or *concrete
+type* is one without free variables. See [Wikipeida][concrete
+wikipedia] for more.
+
+For example, consider this trait:
+
+```rust
+# struct SomeStruct;
+#
+trait Example<A> {
+    fn foo(a: A);
+
+    fn bar<'b>(a: A, b: &'b SomeStruct);
+
+    fn baz();
+
+    fn quux(a: i32, b: Option<i32>);
+}
+```
+
+In it, the trait `Example` has a free variable `A` because it declares it
+itself. Furthermore, `foo` has `A` as a free variable because it uses it as the
```

Comment:
```suggestion
itself. The trait also has `Self` as an implicitly bound variable. Furthermore, `foo` has `A` as a free variable because it uses it as the
```